src/material.js

```diff
@@ -173,24 +173,54 @@ export class Material {
     }
 
     /**
-     * Calculates hash from basis and lattice. Algorithm expects the following:
+     * Calculates the hash for a material.
+     * If `isNonPeriodic` = false
+     * The hash is calculated from the basis and lattice. The algorithm expects the following:
      * - asserts lattice units to be angstrom
      * - asserts basis units to be crystal
      * - asserts basis coordinates and lattice measurements are rounded to hash precision
      * - forms strings for lattice and basis
      * - creates MD5 hash from basisStr + latticeStr + salt
+     *
+     * If `isNonPeriodic` = true
+     * The hash based on the `inchi` string for the material that is returned from getNonPeriodicHashMessage.
+     *
      * @param salt {String} Salt for hashing, empty string by default.
      * @param isScaled {Boolean} Whether to scale the lattice parameter 'a' to 1.
      */
     calculateHash(salt = '', isScaled = false) {
-        const message = this.Basis.hashString + "#" + this.Lattice.getHashString(isScaled) + "#" + salt;
+        let message = this.Basis.hashString + "#" + this.Lattice.getHashString(isScaled) + "#" + salt;
+        if (this.prop('isNonPeriodic')) {
+            const nonPeriodicMessage = this.getNonPeriodicHashMessage();
+            if (nonPeriodicMessage) {
+                message = nonPeriodicMessage;
+            } else {
+                console.error("Non-Periodic hash could not be created. Missing InChI.");
+            }
+        }
         return CryptoJS.MD5(message).toString();
     }
 
+    /**
+     * Gets the `inchi` string from the derived properties and returns it as the message to use for
+     * generating the hash when `isNonPeriodic` = true.
+     * @returns String
+     */
+    getNonPeriodicHashMessage() {
+        const derivedProperties = this.prop('derivedProperties');
+        const inchiString = lodash.isArray(derivedProperties) ? derivedProperties.find(x => x.name === "inchi") : null;
```

Comment:
Did we speak about moving the existing DerivedPropeties logic from webapp here before?